helm/config.go

```diff
@@ -153,32 +149,69 @@ type Node interface {
 	write(enc *Encoder, prefix string)
 }
 
+// NewNode creates a new scalar node.
+func NewNode(value interface{}, modifiers ...NodeModifier) Node {
+	var node Node
+	if value == nil {
+		node = &Scalar{value: "~"}
+	} else if _, ok := value.(Node); ok {
+		node = value.(Node)
+	} else {
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Map:
+			valueOf := reflect.ValueOf(value)
+			mapping := NewMapping()
+			for _, key := range valueOf.MapKeys() {
+				mapping.Add(key.Interface().(string), valueOf.MapIndex(key).Interface())
```

Comment:
I think I still want this to be `mapping.Add(fmt.Sprintf("%s", key.Interface()), …)`. Thoughts?
That won't save you from bad input, but it _will_ save you from a panic. That way the kube validation (on the server end) will deal with the problem.